### PR TITLE
Modify liquidation mechanics

### LINKED ISSUE #1641: [Liquidation improvements] Completely decouple MTM, margin checks and closeout trades
The current design around margins means that:
- closeout trades happen (or are attempted) when a party is in breach of margin requirement and when there is enough volume and when we're not in an auction. 
- we cannot handle (future) frequent batch auctions
- we may reward a party that has a big position (MTM move against them, we want to close out but can't, MTM move for them they're OK again - the network was at risk but party kept profit). 
- the logic in core is tightly coupled in places making moves towards WASM based smart products harder. 

The aim here is:
1. Separate margin checks, mark-to-market and closeout trades into three separate streams that can run at arbitrary frequencies, 
1. Accommodate closeout trades during auctions,
1. Avoid allowing parties with huge position breaching margin level hanging around. 
1. Do this without changing any existing margin calculations. 
1. Do this without changing how loss socialisation works. 

Roughly how: 
- The insurance pool will become "margin account" for a "network party" (think of the network party as a "bad bank"). 
- We will regularly check whether a party is distressed (but doesn't need to be tied to orders / trades etc. only at desired frequency e.g. every block or every 5s). If a party is in breach of margin level their margin is transferred to insurance pool and their position to the "network party". (*)
- We do mark-to-market as usual. If the network party has a position the insurance pool is used for MTM flows. If insurance pool is empty we loss socialise as before. 
- The network party (insurance pool party) will employ a simple heuristic (which later can be WASM but for now we hard code) according to which it will attempt, at a desired intensity, to reduce it's position. 

Insurance party position reduction heuristic:
- if we're in continuous trading place market orders once every desired number of seconds with size the smaller of "half available book volume", "half the position to close out" (unless the position is exactly `1` in which case close out all. 
- if we're in an auction place a limit order - more thinking needed about the size / price; e.g. we don't want to trigger price monitoring or move price too much. 

(*) An added benefit of this is that we may avoid a bunch of close-out trades - imagine network party is long `10` but now it is assigned a position of a party that was short `8`; its new position is `2` ... and a bunch of network trades didn't need to happen.

### PR DESCRIPTION
**Pull request by witgaw**
- Liquidate distressed party's position immediately and move it to the network party.
- Market's insurance pool is market's party margin account.
- Introduce a simple inventory management strategy for the network party.

close https://github.com/vegaprotocol/specs/issues/1641

### CODE REVIEW

**Review by davidsiska-vega — DISMISSED** [review]

**Comment by gordsport** [comment]
The core implementation issue is:

- https://github.com/vegaprotocol/vega/issues/9945

File: protocol/0012-POSR-position_resolution.md [hunk 1]
```diff
@@ -2,167 +2,82 @@
...
+Whenever the network party has a non-zero position it attempts to unload it using an [immediate or cancel](./0014-ORDT-order_types.md) limit order. If the the network has a `long` position it will submit a `sell` order. If it has a short position it will submit a buy order. The size of that order is chosen according to the liquidation strategy which forms a part of the market's configuration. The strategy can be updated at any point whilst market is active with a market change [governance vote](./0028-GOVE-governance.md#2-change-market-parameters).
 
-`// OpenPosition of Network =  -3 +5 -4 +2 = 0`
-`// OpenPosition of Trader1 =  +5 -5 = 0`
-`// OpenPosition of Trader2 = -4 +4 =  0`
-`// OpenPosition of Trader3 =  +2 - 2 = 0`
+Currently only one liquidation strategy is supported and its defined by the following parameters:
 
-#### STEP 5
+- `disposal time step` (min: `1s`, max: `1h`, default: `n=10s`): network attempts to unload its position in a given market every time it goes out of auction and then every `n` seconds as long as market is not in auction mode and while the network's position is not equal to `0`,
```
Comment (EVODelavega) [inline protocol/0012-POSR-position_resolution.md]: Seeing as these are parameters that can be specific to each market, but we want to have default values, should the defaults be hard-coded, or set as network parameters? Can those network parameters be changed via governance, or not?

Comment (witgaw) [reply protocol/0012-POSR-position_resolution.md]: Let's hardcore them, the netparam default for a market parameter ends up just confusing everyone. I'll mention that in the spec.

Comment (EVODelavega) [reply protocol/0012-POSR-position_resolution.md]: @barnabee wonderiing what you make of default values in proposals here

Comment (barnabee) [reply protocol/0012-POSR-position_resolution.md]: I'd suggest no defaults. Make them specify it on the proposal.

Comment (jeremyletang) [reply protocol/0012-POSR-position_resolution.md]: what about market already existing after a protocol upgrade?

Comment (EVODelavega) [reply protocol/0012-POSR-position_resolution.md]: Defaults in protocol upgrade are fine, but that distinction should be made clearer probably -> proposals _must_ specify all 4 parameters, existing markets will default to these default params, which can be updated through governance after protocol upgrade

Comment (witgaw) [reply protocol/0012-POSR-position_resolution.md]: OK, let me know how it's looking now.

**Review by EVODelavega — APPROVED** [review]

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**